### ISSUE TITLE
New rule: condition-paren-newline

### ISSUE THREAD

**Issue by golopot** [eslint/eslint#11636]
**Please describe what the rule should do:**
Enforce consistent line breaks inside of the condition of `IfStatement` and `WhileStatement`.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[x] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

**Options**
The option can be either a string:
- `"never"` - Disallow both newlines.
- `"consistent"`(default) - Enforces either both newlines are present, or both newlines are absent.

Or an object:
```
{
  "singleline":  "consistent" | "none" | "both",
  "multiline": "consistent" | "none" | "both",
}
```
For example:
`{singleline: "none"}` disallows newlines when the contained expression is single-line.
`{multiline: "both"}` requires both newlines when the contained expression is multiline. 

**Examples**
```js
/* eslint condition-paren-newline: ["error", "never"] */

// invalid
if (  
  foo
  && bar
) {}

if (  
  foo
  && bar) {}


// valid
if (foo
  && bar) {}
```

```js
/* eslint condition-paren-newline: ["error", {singleline: "consistent", multiline: "both"}] */

// invalid
if (
  foo
) {}

if ( foo
  && bar ) {}


// valid
if (foo) {}

if (
  foo 
  && bar
) {}
```

**Why should this rule be included in ESLint (instead of a plugin)?**
Similar rules: [function-parens-newline](https://eslint.org/docs/rules/function-paren-newline), [object-curly-newline](https://eslint.org/docs/rules/object-curly-newline), [array-bracket-newline](https://eslint.org/docs/rules/array-bracket-newline).

**Are you willing to submit a pull request to implement this rule?**
Yes.
